src/uct/ib/dc/dc_mlx5.h

```diff
@@ -256,20 +274,23 @@ KHASH_MAP_INIT_INT64(uct_dc_mlx5_fc_hash, uct_dc_mlx5_ep_fc_entry_t);
  * ndci and these stacks are not intersected
  */
 typedef struct {
-    int8_t        stack_top;         /* dci stack top */
-    uint8_t       *stack;            /* LIFO of indexes of available dcis */
-    ucs_arbiter_t arbiter;           /* queue of requests waiting for DCI */
-    int8_t        release_stack_top; /* releasing dci's stack,
-                                        points to last DCI to release
-                                        or -1 if no DCI's to release */
+    int8_t                   stack_top;         /* dci stack top */
+    uct_dc_mlx5_pool_stack_t stack;             /* LIFO of indexes of available dcis */
+    ucs_arbiter_t            arbiter;           /* queue of requests waiting for DCI */
+    int8_t                   release_stack_top; /* releasing dci's stack,
```

Comment:
move this after "stack_top" field, to reduce struct sizer.
pls move new DC structs to ucx_info / type_info and check their sizes - pls post them here and let's review they are reasonable